src/models/modern-bert.cpp

```diff
@@ -0,0 +1,126 @@
+#include "models.h"
+
+template <bool iswa>
+llm_build_modern_bert<iswa>::llm_build_modern_bert(const llama_model & model, const llm_graph_params & params) : llm_graph_context(params) {
+    const int64_t n_embd_head     = hparams.n_embd_head_v;
+    const int64_t n_embd_gqa      = hparams.n_embd_v_gqa();
+    const float rope_theta_global = hparams.rope_freq_base_train;
+    const float rope_theta_local  = hparams.rope_freq_base_train_swa;
+    const uint32_t n_swa_pattern  = hparams.n_swa_pattern;
+
+    GGML_ASSERT(n_embd_head == hparams.n_embd_head_k);
+
+    ggml_tensor * cur = nullptr;
+    ggml_tensor * inpL = nullptr;
+    ggml_tensor * inp_pos = build_inp_pos();
+
+    // construct input embeddings (token, type, position)
+    inpL = build_inp_embd(model.tok_embd);
+    cb(inpL, "inp_embd", -1);
+
+    // embed layer norm
+    inpL = build_norm(inpL, model.tok_norm, nullptr, LLM_NORM, -1);
+    cb(inpL, "inp_norm", -1);
+
+    ggml_tensor * inp_out_ids = build_inp_out_ids();
+
+    auto * inp_attn = build_attn_inp_no_cache();
```

Comment:
What's the reason for using `no_cache` here? Is this standard for all embedding models?